src/main/scala/scalismo/common/interpolation/BarycentricInterpolator.scala

```diff
@@ -33,48 +33,33 @@ object BarycentricInterpolator {
 case class BarycentricInterpolator3D[A: ValueInterpolator](mesh: TetrahedralMesh[_3D])
     extends BarycentricInterpolator[_3D, A] {
 
-  override protected val valueInterpolator: ValueInterpolator[A] = ValueInterpolator[A]
-
-  // TODO: Temporary solution, replace for Milestone M2!
-  private def getTetrahedralMeshCell(p: Point[_3D]): Option[TetrahedralCell] = {
-
-    val numberOfTetrahedrons = mesh.tetrahedralization.tetrahedrons.length
-
-    def isInsideCell(tc: TetrahedralCell): Boolean = mesh.isInsideTetrahedralCell(p, tc)
-    val isInsideCellMemoized = Memoize(isInsideCell, numberOfTetrahedrons)
-
-    var cell: Option[TetrahedralCell] = None
-    var neighbourhood = Set[TetrahedronId]()
+  val meshOps = MeshOperations(mesh)
 
-    while (cell.isEmpty && neighbourhood.size != numberOfTetrahedrons) {
-      if (neighbourhood.isEmpty) {
-        // start from closest vertex point
-        val closestPoint = mesh.pointSet.findClosestPoint(p).id
-        neighbourhood = mesh.tetrahedralization.adjacentTetrahedronsForPoint(closestPoint).toSet
-      } else {
-        // increase neighbourhood
-        neighbourhood =
-          neighbourhood.union(neighbourhood.flatMap(mesh.tetrahedralization.adjacentTetrahedronsForTetrahedron))
-      }
-      val filterResult = neighbourhood.filter(tId => isInsideCellMemoized(mesh.tetrahedralization.tetrahedrons(tId.id)))
-      if (filterResult.nonEmpty) cell = Some(mesh.tetrahedralization.tetrahedrons(filterResult.head.id))
-    }
-    cell
-  }
+  override protected val valueInterpolator: ValueInterpolator[A] = ValueInterpolator[A]
 
   override def interpolate(df: DiscreteField[_3D, UnstructuredPointsDomain[_3D], A]): Field[_3D, A] = {
 
     def interpolateBarycentric(p: Point[_3D]): A = {
-      getTetrahedralMeshCell(p) match {
-        case Some(cell) =>
+
```

Comment:
I would find something like this more clear (mainly because of the last to cases:
```
    cp match {
      case ClosestPointIsVertex(_, _, pid)    => df(pid)
      case ClosestPointOnLine(_, _, pids, bc) => ValueInterpolator[A].blend(df(pids._1), df(pids._2), bc)
      case ClosestPointInTriangleOfTetrahedron(_, _, tetId, triId, bc) => {
        val triangle = mesh.tetrahedralization.tetrahedron(tetId).triangles(triId.id)
        bc.interpolateProperty(df(triangle.ptId1), df(triangle.ptId2), df(triangle.ptId3))
      }
      case ClosestPointInTetrahedron(_, _, tid, bc) => {
        val tetrahedron = mesh.tetrahedralization.tetrahedron(tid)
        bc.interpolateProperty(df(tetrahedron.ptId1), df(tetrahedron.ptId2), df(tetrahedron.ptId3), df(tetrahedron.ptId4))
      }
    }
```

Reply:
I agree, this looks cleaner.